Storing the profile details locally using room database for offline viewing

CodeChamp-SS opened the pull request:
### Description
Used room database with coroutines for storing the profile details for offline viewing. As the user signs in / up, and as soon as they reach the home fragment, the profile details are saved in the database. They are deleted once the user signs out so that if the user signs in / up with any other account then those details are saved and showed when required. I've also updated the kotlin version and the gradle version in the dependencies along with addition of required dependencies of coroutines and room.
If the user clicks on edit profile when they are offline then nothing happens, i.e, they won't be able to edit the profile in order to avoid any crashes which may occur. Once the updates the profile, the changes are saved to the database when the user opens home fragment again.

Fixes #787 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Tested on 2 different emulators and a physical device (Samsung Galaxy J-6) 

![issue 787](https://user-images.githubusercontent.com/52480794/93705693-dea22800-fb3c-11ea-90d0-180383e1316e.gif)

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

CodeChamp-SS committed:
docs : added badges in the documentation

CodeChamp-SS committed:
Merge pull request #1 from CodeChamp-SS/CodeChamp-SS-updating-readme

docs : added badges in the documentation

CodeChamp-SS committed:
docs : updated the link of zulip chat badge

CodeChamp-SS committed:
docs : removed 2 badges

removed open source love and run tests badges

CodeChamp-SS committed:
Merge pull request #2 from CodeChamp-SS/CodeChamp-SS-updating-readme

Code champ ss updating readme

CodeChamp-SS committed:
Merge remote-tracking branch 'upstream/develop' into develop

CodeChamp-SS committed:
feat: storing profile details for offline viewing

CodeChamp-SS committed:
feat: storing profile details for offline viewing

CodeChamp-SS committed:
removing unecessary files

CodeChamp-SS committed:
removing unnecessary changes

CodeChamp-SS committed:
removing unnecessary changes

CodeChamp-SS committed:
removing unnecessary changes

CodeChamp-SS committed:
final removal of unnecessary changes

CodeChamp-SS commented:
@garimasingh128 @vj-codes pls review the pr

garimasingh128 commented:
Just mark the checklist in the PR description!

CodeChamp-SS commented:
@garimasingh128 can u pls tell why the build check is failing ?

garimasingh128 commented:
It shows The command "./gradlew build" exited with 1. Maybe consult with [this](https://stackoverflow.com/questions/30741762/travis-ci-command-gradlew-build-exited-with-1).

@anitab-org/qa-team do you have any idea?

CodeChamp-SS commented:
so is there any problem on my side ? or the server side ?

garimasingh128 commented:
You can cross-check whether you changed some build files or not. Or else let's wait for the review.

CodeChamp-SS commented:
> You can cross-check whether you changed some build files or not. Or else let's wait for the review.
ya i changed some build files as i've also updated the gradle version

CodeChamp-SS commented:
but i don't think that should cause any problem as the app is running fine and the gradle version was very old

CodeChamp-SS commented:
let's wait for the review :-)

CodeChamp-SS commented:
@garimasingh128 any updates from qa-team ?
Shall I remove the changes made to build.gradle files from the pr ? I think then it'll pass the tests

garimasingh128 commented:
Ya you can try. But even if it doesn't, we will look into it during final review. You can also try from your end.

CodeChamp-SS commented:
> Ya you can try. But even if it doesn't, we will look into it during final review. You can also try from your end.

@garimasingh128 any updates on how to fix the build fail ?

epicadk commented:
Can you solve the merge conflicts?

CodeChamp-SS commented:
> Can you solve the merge conflicts?

sure

epicadk commented:
> > Can you solve the merge conflicts?
> 
> sure

actually hang on. idk if this is actually required. I will ask isabel to review again.

vj-codes reviewed (changes requested):
@CodeChamp-SS please resolve the merge conflicts

vj-codes reviewed (changes requested):
@CodeChamp-SS closing this due to inactivity, please let us know if you would like to continue again.
Thank you for your contribution :)

CodeChamp-SS commented:
@vj-codes I think I missed this message😅, I'll resolve the conflicts by next week, I'm a bit busy right now :)

CodeChamp-SS commented:
But ig if this PR is not reviewed and merged soon, then there would be merge conflicts again and again😅, so what do you think of maybe reviewing this PR if you've time?

vj-codes commented:
@CodeChamp-SS no problem, you can resolve them later as well.
I will reopen the PR and put it on hold till then, sounds good?
